tests/main/preseed-core20/task.yaml

```diff
@@ -0,0 +1,162 @@
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+  teardown_fake_store "$STORE_DIR"
+
+execute: |
+  # have snap use the fakestore for assertions (but nothing else)
+  export SNAPPY_FORCE_SAS_URL=http://$STORE_ADDR
+
+  echo "Running pre-seeding"
+  SNAPD_SNAP=$(ls /tmp/snapd*.snap)
+
+  # XXX: confusingly, the name of our test key is actually " (test)".
+  snap prepare-image --preseed --preseed-sign-key=" (test)" --channel=stable --snap=$SNAPD_SNAP model.assertion $PREPARE_IMAGE_DIR
+
+  SYSLABEL=$(ls "$PREPARE_IMAGE_DIR"/system-seed/systems/)
+  PRESEEDAS="$PREPARE_IMAGE_DIR/system-seed/systems/$SYSLABEL/preseed"
+  ARTIFACT="$PREPARE_IMAGE_DIR/system-seed/systems/$SYSLABEL/preseed.tgz"
+
+  echo "Basic check of the generated preseed assertion"
+  if [ ! -f "$PRESEEDAS" ]; then
+    echo "preseed assertion not found"
+    exit 1
+  fi
+  MATCH "type: preseed" < "$PRESEEDAS"
+  MATCH "system-label: $SYSLABEL" < "$PRESEEDAS"
+
+  echo "Examine tasks of the preseeded system"
+  tar -xzf "$ARTIFACT" --strip-components=3 var/lib/snapd/state.json
+  snap debug state state.json --change=1 > tasks.log
+
+  echo "Check that critical tasks of preseeded snapd have expected statuses"
+  # Note, these checks match statuses, but not the order
+  # mark-preseeded task is where snap-preseed stopped, therefore it's in Doing.
+  MATCH "Doing .+ mark-preseeded +Mark system pre-seeded" < tasks.log
+  MATCH "Do .+ mark-seeded +Mark system seeded" < tasks.log
+
+  tar -tzf "$ARTIFACT" > files.log
+
+  echo "Check that critical files are present in the artifact"
+  # The list of files is not exhaustive as it would be annoying to maintain; all
+  # the critical files should be checked though.
+  MATCH "^etc/systemd/user/sockets.target.wants/snapd.session-agent.socket" < files.log
+  MATCH "^etc/systemd/user/snapd.session-agent.socket" < files.log
+  MATCH "^etc/systemd/user/snapd.session-agent.service" < files.log
+
+  MATCH "^etc/dbus-1/system.d/wpa_supplicant.conf" < files.log
+  MATCH "^etc/dbus-1/system.d/snapd.system-services.conf" < files.log
+
+  MATCH "^var/lib/snapd/dbus-1/services/io.snapcraft.SessionAgent.service" < files.log
+  MATCH "^var/lib/snapd/dbus-1/services/io.snapcraft.Launcher.service" < files.log
+  MATCH "^var/lib/snapd/dbus-1/services/io.snapcraft.Settings.service" < files.log
+
+  MATCH "^etc/systemd/system/cloud-final.service.wants/snapd.seeded.service" < files.log
+  MATCH "^etc/systemd/system/snapd.snap-repair.service" < files.log
+  MATCH "^etc/systemd/system/snapd.system-shutdown.service" < files.log
+  MATCH "^etc/systemd/system/snapd.core-fixup.service" < files.log
+  MATCH "^etc/systemd/system/snap-pc-.*.mount" < files.log
+  MATCH "^etc/systemd/system/usr-lib-snapd.mount" < files.log
+  MATCH "^etc/systemd/system/snapd.spread-tests-run-mode-tweaks.service" < files.log
+  MATCH "^etc/systemd/system/dbus-fi.w1.wpa_supplicant1.service" < files.log
+  MATCH "^etc/systemd/system/snap-pc.*.mount" < files.log
```

Comment:
snap-pc is matched twice here, as this catches both the pc snap and the pc-kernel snap, given that `-` is exported funnily, you could use `MATCH "^etc/systemd/system/snap-pc.*kernel-.*.mount" < files.log` maybe?

Reply:
Done.